src/service/message.js

```diff
@@ -1,85 +1,133 @@
 const fs = require('fs');
+const readline = require('readline');
 
-const addMessage = (message) => {
-  let lineBreak = '';
+const createInterface = () => readline.createInterface({
+  input: fs.createReadStream('log.txt'),
+  output: process.stdout,
+  terminal: false
+});
 
-  if (fs.existsSync('log.txt') && fs.readFileSync('log.txt', 'utf8') !== '') {
-    lineBreak = '\n';
-  }
+const addMessage = (newMessage) => new Promise((resolve, rejected) => {
+  let lineBreak = '';
 
-  fs.appendFileSync('log.txt', `${lineBreak}${message}`);
+  if (fs.existsSync('log.txt')) {
+    const rl = createInterface();
+    let newId = 0;
 
-  return message;
-};
+    rl.on('line', line => {
+      const { id, message } = JSON.parse(line);
+      if (line) {
+        fs.appendFileSync('log1.txt', `${lineBreak}${JSON.stringify({ id, message })}`);
+      }
+      if (!lineBreak) { lineBreak = '\n'; }
+
+      newId = id;
+    });
+
+    rl.on('close', () => {
+      fs.unlinkSync('log.txt');
+      fs.appendFileSync('log1.txt', `${lineBreak}${JSON.stringify({ id: ++newId, message: newMessage })}`);
+      fs.renameSync('log1.txt', 'log.txt');
+      resolve(newMessage);
+    });
+  } else {
+    rejected(new Error('file not exiest'));
+  }
+});
 
 const findAll = () => {
   if (fs.existsSync('log.txt')) {
     const fileData = fs.readFileSync('log.txt', 'utf8');
 
     if (fileData) {
-      return fileData.split('\n');
+      return fileData;
```

Comment:
is it array in `fileData`? I believe it's list of stringified JSONs separated by `\n`.